sklearn/ensemble/_hist_gradient_boosting/tests/test_grower.py

```diff
@@ -567,3 +567,69 @@ def test_ohe_equivalence(min_samples_leaf, n_unique_categories, target):
         assert predictor.get_max_depth() < predictor_ohe.get_max_depth()
 
     np.testing.assert_allclose(preds, preds_ohe)
+
+
+def test_grower_interaction_constraints():
+    """Check that grower respects interaction constraints."""
+    n_features = 6
+    interaction_cst = [{0, 1}, {1, 2}, {3, 4, 5}]
+    n_samples = 10
+    n_bins = 6
+    root_feature_splits = []
+
+    def get_all_children(node):
+        res = []
+        if node.is_leaf:
+            return res
+        for n in [node.left_child, node.right_child]:
+            res.append(n)
+            if not n.is_leaf:
+                res.extend(get_all_children(n))
+        return res
+
+    for seed in range(20):
```

Comment:
Is this a candidate for using `global_random_seed`?

Reply:
Not at this place. The loop over 20 seeds is needed to ensure that every feature was split on at least once in `TreeGrower` (which is tested at the end of the test).

With `global_random_seed` instead of the loop, it not possible (or hard) to ensure that.

Reply:
```suggestion
    # Ensure that every feature was split on at least once in `TreeGrower`
    # on different trials.
    for seed in range(20):
```